frontend/vite-project/src/components/card-component/ProductCard.tsx

```diff
@@ -0,0 +1,62 @@
+import { useState, useEffect } from 'react';
+import { ProductCardProps } from '../../types/types';
+import { QuantityControl } from '../quantity-control/QuantityControl';
+import './productCard.css';
+
+const ProductCard: React.FC<ProductCardProps> = ({
+  imageUrl,
+  seedName,
+  price,
+  quantity,
+  onQuantityChange,
+  onAddToCart
+}) => {
+  const [counter, setCounter] = useState(quantity);
+
+  useEffect(() => {
+    setCounter(quantity);
+  }, [quantity]);
+
+  const handleDecrement = () => {
+    const newCount = Math.max(counter - 1, 0);
+    setCounter(newCount);
+    onQuantityChange?.(newCount);
+  };
+
+  const handleIncrement = () => {
+    const newCount = counter + 1;
+    setCounter(newCount);
```

Comment:
```suggestion
    setCounter((prevState) => (prevState -1 === 0 ? 1: prevState -1));
```
A suggestion so the counter/quantity doesn’t go below 1.
Send me a message if it’s confusing.